src/App.tsx

```diff
@@ -1,26 +1,208 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  creatTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { SortField } from './types/SortField';
+import { filterTodos } from './utils/helpers.ts/filterTodos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/Todolist';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState(SortField.All);
+  const [title, setTitle] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingIds, setProcessingIds] = React.useState<number[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const showErrorMessage = (errorMessage: string) => {
+    setError(errorMessage);
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  const filteredTodos = filterTodos(todos, filter);
+  const handelFilter = (filterValue: SortField) => {
+    setFilter(filterValue);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
```

Comment:
Create variable for `title.trim()`, you do it at least twice

Reply:
> Create variable for `title.trim()`, you do it at least twice

Good point